verilog/tools/ls/verible-lsp-adapter.cc

```diff
@@ -48,22 +48,32 @@ static verible::lsp::Diagnostic ViolationToDiagnostic(
 }
 
 std::vector<verible::lsp::Diagnostic> CreateDiagnostics(
-    const BufferTracker &tracker) {
+    const BufferTracker &tracker, int message_limit) {
   // Diagnostics should come from the latest state, including all the
   // syntax errors.
   const ParsedBuffer *const current = tracker.current();
   if (!current) return {};
-  // TODO: files that generate a lot of messages will create a huge
-  // output. So we limit the messages here.
-  // However, we should work towards emitting them around the last known
-  // edit point in the document as this is what the user sees.
-  static constexpr int kMaxMessages = 100;
   const auto &rejected_tokens = current->parser().GetRejectedTokens();
   auto const &lint_violations =
       verilog::GetSortedViolations(current->lint_result());
   std::vector<verible::lsp::Diagnostic> result;
   int remaining = rejected_tokens.size() + lint_violations.size();
-  if (remaining > kMaxMessages) remaining = kMaxMessages;
+
+  // TODO: files that generate a lot of messages will create a huge
+  // output. So we limit the messages here if "message_limit" is set.
+  //
+  // We might consider emitting them around the last known
+  // edit point in the document as this is what the user sees (if we get
+  // individual edits, not full files pushed).
+  //
+  // TODO(hzeller): to limit repetition, maybe limit the number of messages
+  // coming from the _same_ source if we have a "message_limit". So for
+  // instance, don't complain on every single line not to use tabs as
+  // indentation.
+  if (message_limit > 0 && remaining > message_limit) {
```

Comment:
Below you pass -1 to signal no-limit, so should this be `>= 0`?

Reply:
Done.